core/os/thread_safe.h

```diff
@@ -31,6 +31,8 @@
 #ifndef THREAD_SAFE_H
 #define THREAD_SAFE_H
 
+#include "core/os/mutex.h" // IWYU pragma: keep // Used in macro.
```

Comment:
CC @Repiteo, I took this format from #100918.

Does that work for `clangd` too and not just `include-what-you-use`?

Reply:
It indeed worked for `clangd`, so much so that it's part of their guidelines:

https://clangd.llvm.org/guides/include-cleaner#scenarios-and-solutions